src/log/log.cr

```diff
@@ -33,33 +33,41 @@ class Log
     @backend = value
   end
 
-  {% for method, severity in {
-                               trace:  Severity::Trace,
-                               debug:  Severity::Debug,
-                               info:   Severity::Info,
-                               notice: Severity::Notice,
-                               warn:   Severity::Warn,
-                               error:  Severity::Error,
-                               fatal:  Severity::Fatal,
-                             } %}
-    # Logs a message if the logger's current severity is lower or equal to `{{severity}}`.
-    def {{method.id}}(*, exception : Exception? = nil)
-      severity = Severity.new({{severity}})
+  {% for severity in Log::Severity.constants %}
+    {% method = severity.id.underscore %}
+
+    # Logs a message if the logger's current severity is lower or equal to
+    # `Severity::{{ severity }}`.
+    #
+    # The block is not called unless the current severity level would emit a
+    # message.
+    #
+    # Blocks which return nil do not emit anything:
+    #
+    # ```
+    # Log.{{ method }} {
+    #   if false
+    #     "Nothing will be logged."
+    #   end
+    # }
+    # ```
+    def {{method}}(*, exception : Exception? = nil)
```

Comment:
@straight-shoota Good call. I added an example here as well. The documentation here felt like it could be improved a bit by restructuring this macro iteration in order to get the name of the severity instead of just the number:

Before:
<img width="703" alt="image" src="https://user-images.githubusercontent.com/208647/163596724-50acf2a2-ed33-46a0-baca-645827bf66ef.png">

After:
<img width="737" alt="image" src="https://user-images.githubusercontent.com/208647/163596866-813ebd47-0c56-455b-8bc5-08cf9554d3a1.png">

Reply:
`do / end` would fit better. Multiline using `{ }`, while valid, is more so used for single line blocks. E.g. `Log.warn { "Nothing will be logged" if false }` versus:

```cr
Log.warn do
  if false
    "Nothing will be logged"
  end
end
```

Reply:
It's a stylistic debate. `{ }` is perfectly fine. It's often used to indicate that the block's output value matters.